internal/versions/deck_test.go

```diff
@@ -3,25 +3,13 @@ package versions_test
 import (
 	"testing"
 
-	"github.com/blang/semver/v4"
 	"github.com/stretchr/testify/require"
 
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/versions"
 )
 
 func TestDeckFileFormat(t *testing.T) {
-	t.Run("Kong version >= 3.0", func(t *testing.T) {
-		actualDeckFormat := versions.DeckFileFormat(semver.Version{Major: 3, Minor: 0})
-		require.Equal(t, "3.0", actualDeckFormat)
-	})
-
-	t.Run("Kong version >= 3.0 - other than 3.0", func(t *testing.T) {
-		actualDeckFormat := versions.DeckFileFormat(semver.Version{Major: 3, Minor: 5})
-		require.Equal(t, "3.0", actualDeckFormat)
-	})
-
-	t.Run("Kong version < 3.0", func(t *testing.T) {
-		actualDeckFormat := versions.DeckFileFormat(semver.Version{Major: 2, Minor: 9})
-		require.Equal(t, "1.1", actualDeckFormat)
+	t.Run("Kong Getaway version supported by KIC >=3.0.0 should has 3.0 config format", func(t *testing.T) {
```

Comment:
Seeing this it seems that we can also adjust the logic of https://github.com/Kong/kubernetes-ingress-controller/blob/8d27fd1fa351be9a5e80204a37217a93bea9a5b4/internal/versions/deck.go#L6-L11 since anything older than 3.4 is not going to be supported.

Reply:
Yeah, I spotted this and started to do it in a separate PR, but it fits here perfectly. I'm doing it now